packages/amplication-plugin-api/src/plugin/plugin.resolver.ts

```diff
@@ -9,29 +9,46 @@ import { PluginService } from "./plugin.service";
 import { Public } from "../decorators/public.decorator";
 import { GraphQLError } from "graphql";
 import { PluginVersion } from "../pluginVersion/base/PluginVersion";
+import { PluginVersionService } from "../pluginVersion/pluginVersion.service";
+import { ProcessedPluginVersions } from "./plugin.types";
+import { PluginVersionFindManyArgs } from "../pluginVersion/base/PluginVersionFindManyArgs";
 
 @graphql.Resolver(() => Plugin)
 @common.UseGuards(GqlDefaultAuthGuard, gqlACGuard.GqlACGuard)
 export class PluginResolver extends PluginResolverBase {
   constructor(
     protected readonly service: PluginService,
     @nestAccessControl.InjectRolesBuilder()
-    protected readonly rolesBuilder: nestAccessControl.RolesBuilder
+    protected readonly rolesBuilder: nestAccessControl.RolesBuilder,
+    protected readonly pluginVersionService: PluginVersionService
   ) {
     super(service, rolesBuilder);
   }
 
   @Public()
-  @graphql.Query(() => [Plugin], { nullable: true })
-  async githubPlugins(): Promise<Plugin[]> {
+  @graphql.Mutation(() => [Plugin], { nullable: true })
+  async processPluginCatalog(): Promise<ProcessedPluginVersions[]> {
     try {
       const amplicationPlugins = await this.service.githubCatalogPlugins();
       if (
         Object.prototype.toString.call(amplicationPlugins) === "[object String]"
-      )
+      ) {
```

Comment:
personally, I don't think we should have curly bracket in one if line, but if we agree for that we should do it all over as a best practice.